ldk/go/windowClient.go

```diff
@@ -0,0 +1,135 @@
+package ldk
+
+import (
+	"context"
+	"errors"
+	"io"
+
+	"github.com/open-olive/loop-development-kit/ldk/go/proto"
+)
+
+type WindowClient struct {
+	client  proto.WindowClient
+	session *Session
+}
+
+func (w *WindowClient) ActiveWindow() (WindowInfo, error) {
+	ctx, cancel := context.WithTimeout(context.Background(), grpcTimeout)
+	defer cancel()
+
+	resp, err := w.client.WindowActiveWindow(ctx, &proto.WindowActiveWindowRequest{})
+	if err != nil {
+		return WindowInfo{}, err
+	}
+
+	return WindowInfo{
+		Title:  resp.Window.GetTitle(),
+		Path:   resp.Window.GetPath(),
+		PID:    int(resp.Window.GetPid()),
+		X:      int(resp.Window.GetX()),
+		Y:      int(resp.Window.GetY()),
+		Width:  int(resp.Window.GetWidth()),
+		Height: int(resp.Window.GetHeight()),
+	}, nil
+}
+
+func (w *WindowClient) ListenActiveWindow(ctx context.Context, handler ListenActiveWindowHandler) error {
+	stream, err := w.client.WindowActiveWindowStream(ctx, &proto.WindowActiveWindowStreamRequest{})
+	if err != nil {
+		return err
+	}
+
+	go func() {
+		for {
+			resp, err := stream.Recv()
+			if err == io.EOF {
+				break
```

Comment:
Just curious (I'm still learning Go), but how does this goroutine ever stop? Does it stop here when the active window is closed I guess?

Reply:
This goroutine will stop whenever it's closed. IIRC the `stream.Recv()` function blocks until the stream sends something, and if the stream is closed it'll send the `io.EOF` error.

Otherwise, it'll send changes as the Active Window information changes.

Reply:
Pretty sure that's correct. But to add to it - the way the stream is closed is by canceling the `ctx` (which is passed into the creation of the `stream`). So however/whenever `ctx` is closed is the source of "how" this is controlled